qiita_pet/handlers/download.py

```diff
@@ -77,6 +46,149 @@ def _generate_files(self, header_name, accessions, filename):
         self.write(text)
         self.finish()
 
+    def _list_dir_files_nginx(self, dirpath):
+        """Generates a nginx list of files in the given dirpath for nginx
+
+        Parameters
+        ----------
+        dirpath : str
+            Path to the directory
+
+        Returns
+        -------
+        list of (str, str, str)
+            The path information needed by nginx for each file in the
+            directory
+        """
+        basedir = get_db_files_base_dir()
+        basedir_len = len(basedir) + 1
+        to_download = []
+        for dp, _, fps in walk(dirpath):
+            for fn in fps:
+                fullpath = join(dp, fn)
+                spath = fullpath
+                if fullpath.startswith(basedir):
+                    spath = fullpath[basedir_len:]
+                to_download.append((fullpath, spath, spath))
+        return to_download
+
+    def _list_artifact_files_nginx(self, artifact):
+        """Generates a nginx list of files for the given artifact
+
+        Parameters
+        ----------
+        artifact : qiita_db.artifact.Artifact
+            The artifact to retrieve the files
+
+        Returns
+        -------
+        list of (str, str, str)
+            The path information needed by nginx for each file in the artifact
+        """
+        basedir = get_db_files_base_dir()
+        basedir_len = len(basedir) + 1
+        to_download = []
+        for i, (fid, path, data_type) in enumerate(artifact.filepaths):
+            # ignore if tgz as they could create problems and the
+            # raw data is in the folder
+            if data_type == 'tgz':
+                continue
+            if data_type == 'directory':
```

Comment:
Would it be better to ask the filesystem if the path is a directory?

Reply:
done